java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java

```diff
@@ -57,25 +60,48 @@
 
     private static final String TEST_DATA = "EPISODEII_IS_BEST_EPISODE";
 
+    private Boolean bucketsEnabled = false;
+
     private MantaClient mantaClient;
 
     private String testPathPrefix;
 
+    private String bucketsTestPathPrefix;
+
     @BeforeClass
-    @Parameters({"usingEncryption"})
-    public void beforeClass(@Optional Boolean usingEncryption) throws IOException {
+    @Parameters({"usingEncryption", "testType"})
+    public void beforeClass(@Optional Boolean usingEncryption,
+                            @Optional String testType) throws IOException {
 
         // Let TestNG configuration take precedence over environment variables
         ConfigContext config = new IntegrationTestConfigContext(usingEncryption);
+        String testName = this.getClass().getSimpleName();
 
         mantaClient = new MantaClient(config);
-        testPathPrefix = IntegrationTestConfigContext.generateBasePath(config, this.getClass().getSimpleName());
-        mantaClient.putDirectory(testPathPrefix, true);
+
+        if(testType != null && testType.equals("buckets")) {
+            if(IntegrationTestHelper.verifyBucketsSupport(config, mantaClient)) {
+                bucketsEnabled = true;
+                bucketsTestPathPrefix = IntegrationTestHelper.setupBucketsPath(config, testName);
+                //When the buckets api is ready create the bucket here.
+                testPathPrefix = IntegrationTestHelper.setupBucketObjectBasePath(bucketsTestPathPrefix);
+            } else {
+                throw new SkipException("Buckets operations not supported by the server");
+            }
+        } else {
+            testPathPrefix = IntegrationTestConfigContext.generateBasePath(config, testName);
+            mantaClient.putDirectory(testPathPrefix, true);
+        }
+
     }
 
     @AfterClass
     public void afterClass() throws IOException {
-        IntegrationTestConfigContext.cleanupTestDirectory(mantaClient, testPathPrefix);
+        if(BooleanUtils.isTrue(bucketsEnabled)) {
```

Comment:
Could the `IntegrationTestHelper` have a more generic method that would do the right thing based on the `testType`? Is there a way in which we could lessen duplication?